uint32[] node_ids

float64[] weights